stackslib/src/net/api/gettransaction.rs

```diff
@@ -0,0 +1,197 @@
+use stacks_common::types::StacksPublicKeyBuffer;
+use stacks_common::util::hash::{to_hex, Hash160, Sha256Sum};
+
+use crate::burnchains::affirmation::AffirmationMap;
+use crate::burnchains::Txid;
+use crate::chainstate::burn::db::sortdb::SortitionDB;
+use crate::chainstate::stacks::db::StacksChainState;
+use crate::core::mempool::MemPoolDB;
+use crate::net::http::{
+    parse_json, Error, HttpNotFound, HttpRequest, HttpRequestContents, HttpRequestPreamble,
+    HttpResponse, HttpResponseContents, HttpResponsePayload, HttpResponsePreamble, HttpServerError,
+};
+use crate::net::httpcore::{
+    request, HttpPreambleExtensions, RPCRequestHandler, StacksHttpRequest, StacksHttpResponse,
+};
+use crate::net::p2p::PeerNetwork;
+use crate::net::{Error as NetError, StacksNodeState};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct TransactionResponse {
+    pub block_hash: String,
+    pub tx: String,
+}
+
+#[derive(Clone)]
+pub struct RPCGetTransactionRequestHandler {
+    pub txid: Option<Txid>,
+}
+impl RPCGetTransactionRequestHandler {
+    pub fn new() -> Self {
+        Self { txid: None }
+    }
+}
+
+/// Decode the HTTP request
+impl HttpRequest for RPCGetTransactionRequestHandler {
+    fn verb(&self) -> &'static str {
+        "GET"
+    }
+
+    fn path_regex(&self) -> Regex {
+        Regex::new(r#"^/v3/transactions/(?P<txid>[0-9a-f]{64})$"#).unwrap()
+    }
+
+    fn metrics_identifier(&self) -> &str {
+        "/v3/transactions/:txid"
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        captures: &Captures,
+        query: Option<&str>,
+        _body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() != 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected 0-length body for GetTransaction".to_string(),
+            ));
+        }
+
+        let txid = request::get_txid(captures, "txid")?;
+        self.txid = Some(txid);
+
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCGetTransactionRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.txid = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let txid = self
+            .txid
+            .take()
+            .ok_or(NetError::SendError("`txid` no set".into()))?;
+
+        let txinfo_res =
+            node.with_node_state(|_network, _sortdb, chainstate, mempool, _rpc_args| {
+                let index_block_hashes = match chainstate.chainstate_tx_begin() {
```

Comment:
This does not need to be a transaction, and _should not be_, since it could stall block-processing (or cause the P2P thread to be stalled by block-processing).